WikiWatershed/model-my-watershed
# Add New Layerpicker

## Overview
This PR aims to take us 80% of the way to having a new, wired up layer picker. 

- The first commit is Jeff's prototype
- The second commit reorganizes how we store the layer config. Originally it was a long list of all the layers, and used boolean flags to determine what category they belonged to. Now they have been broken up into a dictionary. Each key holds a list with all the layers of that type (eg streams, basemap, etc)
- The third commit adds the new layer picker control as its own set of backbone views (instead of a leaflet control). The building of layers from the config is now handled by a layers backbone model.

#### Todo
- [x]  ~Opacity Control~ PR #1667 
- [x] ~Observations Deferred Tab (don't click the final tab right now...)~ PR #1673 
- [x] ~Legends~ issue #1674
- [x] Testing ITSI Interactive Screenshot (changed some code related to fixing the google layer bug #1640)
- [x] Compare view bug (when you come back from the compare view, you get stuck with a single tile until you switch layers/zooms. See screenshot below)
<img width="960" alt="screen shot 2017-01-17 at 6 24 03 pm" src="https://cloud.githubusercontent.com/assets/7633670/22070556/75f6ff0e-dd6a-11e6-8b44-e529a748324f.png">

## Demo
![layerpickeruno](https://cloud.githubusercontent.com/assets/7633670/22070478/300c3144-dd6a-11e6-904f-559e76cc95b7.gif)

## Testing Instructions
- Pull this branch and bundle up
- Add and remove layers. 
- Confirm certain layers (like PA municipalities) are disabled when you are too zoomed out, or the viewport is outside the US. 
- Test out the Select By Boundary

Connects #1652
